common/include/pcl/pcl_macros.h

```diff
@@ -448,3 +448,30 @@ aligned_free(void* ptr)
 #else
   #define PCL_IF_CONSTEXPR(x) if (x)
 #endif
+
+// [[unlikely]] can be used on any conditional branch, but __builtin_expect is restricted to the evaluation point
+// This makes it quite difficult to create a single macro for switch and while/if
+/**
+ * @def PCL_CONDITION_UNLIKELY
+ * @brief marks a conditional branch as more likely than the other
+ * \code{.cpp}
+ * if PCL_CONDITION_UNLIKELY(x == 0) { return; } else { throw std::runtime_error("some error"); }
+ * //
+ * while PCL_CONDITION_UNLIKELY(is_complete) { continue; }  // busy wait, with minimal chances of waiting
+ * \endcode
+ * @note This can't be used with switch statements. Also, prefer using PCL_{IF,WHILE}_UNLIKELY
+ * @details Most compilers assume that the condition would evaluate to true in if and while loops.
```

Comment:
Do you have a reference for this? Sounds plausible, but would be interesting to know if there is some article/documentation about this

Reply:
Err... I don't think there's documentation, just knowledge in my head. I verified this with a couple of compilers (https://godbolt.org/z/11W1z9131), but can't find a reference doc for the compilers

Reply:
Couple = 10, before I wrote this 😆